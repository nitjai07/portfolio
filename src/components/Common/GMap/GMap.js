import React from 'react';
import GoogleMapReact from 'google-map-react';
import PropTypes from 'prop-types';

import Typewriter from '../Typewriter/Typewriter';

const AnyReactComponent = () => (
  <div className="mapOverText">
    Nitish Jain <br />
    Aishwarya Opulance Apartment, Marathahalli <br />
    Bangalore- 560037 <br /> <br />
    <span>@ </span>
    <a href="mailto:nitishjain2224@gmail.com" className="linkdefault" target="_top">
      <Typewriter idValue="mainOnMap" typeTxt="nitishjain2224@gmail.com" />
    </a>
  </div>
);

function SimpleMap(props) {
  return (
    // Important! Always set the container height explicitly
    <div className={props.customClass}>
      <GoogleMapReact
        bootstrapURLKeys={{ key: 'AIzaSyAKIRNlT44mTTMRaL0Wm3npkMf-Deyasjw' }}
        defaultCenter={{ lat: props.latitude, lng: props.longitude }}
        defaultZoom={props.zoom}
      >
        <AnyReactComponent lat={props.latitude} lng={props.longitude} />
      </GoogleMapReact>
    </div>
  );
}

SimpleMap.propTypes = {
  customClass: PropTypes.string,
  latitude: PropTypes.number.isRequired,
  longitude: PropTypes.number.isRequired,
  zoom: PropTypes.number
};

SimpleMap.defaultProps = {
  customClass: '',
  zoom: 8
};

export default SimpleMap;
