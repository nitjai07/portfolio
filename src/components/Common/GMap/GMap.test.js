import React from 'react';
import { shallow } from 'enzyme';
import GMap from './GMap';

test('Heading Component renders correctly', () => {
  const component = shallow(<GMap latitude={12.123} longitude={32.21} />);
  expect(component).toMatchSnapshot();
});
