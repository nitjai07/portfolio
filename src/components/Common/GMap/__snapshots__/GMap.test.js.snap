// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className=""
>
    <GoogleMap
        bootstrapURLKeys={
            Object {
                "key": "AIzaSyAKIRNlT44mTTMRaL0Wm3npkMf-Deyasjw",
              }
        }
        debounced={true}
        defaultCenter={
            Object {
                "lat": 12.123,
                "lng": 32.21,
              }
        }
        defaultZoom={8}
        distanceToMouse={[Function]}
        googleMapLoader={[Function]}
        heatmap={Object {}}
        heatmapLibrary={false}
        hoverDistance={30}
        layerTypes={Array []}
        options={[Function]}
        style={
            Object {
                "height": "100%",
                "margin": 0,
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
        }
        yesIWantToUseGoogleMapApiInternals={false}
    >
        <AnyReactComponent
            lat={12.123}
            lng={32.21}
        />
    </GoogleMap>
</div>,
  "nodes": Array [
    <div
      className=""
>
      <GoogleMap
            bootstrapURLKeys={
                  Object {
                        "key": "AIzaSyAKIRNlT44mTTMRaL0Wm3npkMf-Deyasjw",
                      }
            }
            debounced={true}
            defaultCenter={
                  Object {
                        "lat": 12.123,
                        "lng": 32.21,
                      }
            }
            defaultZoom={8}
            distanceToMouse={[Function]}
            googleMapLoader={[Function]}
            heatmap={Object {}}
            heatmapLibrary={false}
            hoverDistance={30}
            layerTypes={Array []}
            options={[Function]}
            style={
                  Object {
                        "height": "100%",
                        "margin": 0,
                        "padding": 0,
                        "position": "relative",
                        "width": "100%",
                      }
            }
            yesIWantToUseGoogleMapApiInternals={false}
      >
            <AnyReactComponent
                  lat={12.123}
                  lng={32.21}
            />
      </GoogleMap>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <SimpleMap
      customClass=""
      latitude={12.123}
      longitude={32.21}
      zoom={8}
/>,
    "_forcedUpdate": false,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className=""
>
      <GoogleMap
            bootstrapURLKeys={
                  Object {
                        "key": "AIzaSyAKIRNlT44mTTMRaL0Wm3npkMf-Deyasjw",
                      }
            }
            debounced={true}
            defaultCenter={
                  Object {
                        "lat": 12.123,
                        "lng": 32.21,
                      }
            }
            defaultZoom={8}
            distanceToMouse={[Function]}
            googleMapLoader={[Function]}
            heatmap={Object {}}
            heatmapLibrary={false}
            hoverDistance={30}
            layerTypes={Array []}
            options={[Function]}
            style={
                  Object {
                        "height": "100%",
                        "margin": 0,
                        "padding": 0,
                        "position": "relative",
                        "width": "100%",
                      }
            }
            yesIWantToUseGoogleMapApiInternals={false}
      >
            <AnyReactComponent
                  lat={12.123}
                  lng={32.21}
            />
      </GoogleMap>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SimpleMap
    customClass=""
    latitude={12.123}
    longitude={32.21}
    zoom={8}
/>,
}
`;
